```
{
    "status": "FAIL", 
    "group": "default", 
    "message": "AssertionError: assert 'iVBORw0KGgoA...JRU5ErkJggg==' == 'iVBORw0KGgoAA...AAElFTkSuQmCC'", 
    "stack": "session = <Session 3cec4c58-be13-4c03-9a1a-f5bfb9c8bfc4>\nurl = <function url at 0x7f6476622c80>, domain = 'alt'\n\n    @pytest.mark.parametrize(\"domain\", [\"\", \"alt\"], ids=[\"same_origin\", \"cross_origin\"])\n    def test_source_origin(session, url, domain):\n        session.url = inline(\"\"\"{0}{1}\"\"\".format(DEFAULT_CSS_STYLE, DEFAULT_CONTENT))\n    \n        response = take_screenshot(session)\n        reference_screenshot = assert_success(response)\n        assert png_dimensions(reference_screenshot) == viewport_dimensions(session)\n    \n        iframe_content = \"<style>body {{ margin: 0; }}</style>{}\".format(DEFAULT_CONTENT)\n        session.url = inline(\"\"\"{0}{1}\"\"\".format(\n            DEFAULT_CSS_STYLE, iframe(iframe_content, domain=domain)))\n    \n        response = take_screenshot(session)\n        screenshot = assert_success(response)\n        assert png_dimensions(screenshot) == viewport_dimensions(session)\n    \n>       assert screenshot == reference_screenshot\nE       AssertionError: assert 'iVBORw0KGgoA...JRU5ErkJggg==' == 'iVBORw0KGgoAA...AAElFTkSuQmCC'\nE         - iVBORw0KGgoAAAANSUhEUgAABAAAAALkCAIAAAAeetB6AAAr50lEQVR4nO3dfbzX8/348U/ZsLGNLoRfjK8pXUxWbIj4uvyua7ou5SITc5FRKulKoisWUhoNqXRlpatZpcI3hg2NqMM2XxmTkiUX6eL83p9zOqfPOXXO0almnef9/odbn/f7836d99vzn/fj8/mczymXnZ2dAgAAYhAAAAAQiAAAAIBABAAAAAQiAAAAIBABAAAAgQgAAAAIRAAAAEAgRQZAuXLJfyjjihg+AABlVnEBUMQeyggjBgAISADEZcQAAAEJgLiMGAAgIAEQlxEDAAQkAOIyYgCAgARAXEYMABCQAIjLiAEAAhIAcRkxAEBAAiAuIwYACEgAxGXEAAABCYC4jBgAICABEJcRAwAEJADiMmIAgIAEQFxGDAAQkACIy4gBAAISAHEZMQBAQAIgLiMGAAhIAMRlxAAAAQmAuIwYACCgXQqAz96eM2rYfZPmvZT1wbrN36n0wxp165/VsHnrlucfX/nbye5S2zCt9UGtpqZaTf1kSsv9ksfbKfEJpZasfOiU1v8sYtlk7x76uYk9uviOfJ0RAwBQxpQ6ALZ88PtuFw1Z16xH15b1q1X57pZ172e9+Puxt/UdtWTdhVN2/RY2uRsu5kY8UeITSqfEZUt8wq7Yo4tvp6QRAwBQBpUyALb8bUzT9m/86skRZx9ULnmYb/3irnUbf3D7ml2+hS3xVrjEJ5ROicuW+IRdsUcX307xIwYAoEwqXQBseOb6E4fWnjfr8sMK3P4Xlv3pa+P79Ro++ZkVa1IVqzdo0+2OARf9+HvpQ5I73UIfdym0JXlY8FZ4/bIJ/XoMeezp9FLHnXXZbUNqPHrKE5lP2Co5MGedCUs7rejRZ+xTb64uV7nWeVcOHdnr7Crlk/0ln9WXyb9yla98+dz3Hjh/3+Sf+ZLnFDixoldL++KdeaMGDv7tk3/627++c9TPLrh+yOArTsxIpq97UXtGsSMGAKBsKlUAbHr5lhOv/f6UZ26qtk/yqCgbV9zXuPHEmgNG3NikziHZq5bOGt6135sd5sy+unrOLwgUvpMuuCXz38kPfGtMs0bjqvVPL1Wl3OrXn7zv5hEzViytNXT11icUkBxbqWfWaadcdlP/jqceud/a1yZ3a9t78+2vT2xZodzOn1UhBZ9Q8moVu+8zYtpd7Y8/+PPlM27p2HvDwJcfavKDZNfOXtQeUNyIAQAoo0oVABueuOiIh37+9owO308eFOVfMzrVHXPa03OuqJr7wnsqtWXlmIZnLrny5XHN03fAyc1xxp10WuaWzH+n1s26tN7oUxbN3rZU9keT2tTonLr/49wnFJQcW+H6H0xb/sDPD0weJTZnDT3tnL/d+vb95+6782dVSIEnlLRaQdmrf9vkxBevz7r/nPR7Cjt5UXtAcSMGAKCMKn0A/M/bMy4qJgA2Pn1dzYHVF827Ov/+Nn13PPLcs7L6vnHPGcnr4wXupHNkbsn898Znuta8tVrBpbI/Gtvw2D90/jDnCYWkj32k8d9ndTooeZDj80ktqk5v98HkluV3/qwKyXxCiddYSHLsYVNbJ6eRPnYnL2oPKG7EAACUUaUKgE1/7l3v2u9PebZH9aI/ArRhevuq45u+83jbA5IHeT6b1OKoWR3fm9A8ucFN7obz76RzZW4p8O8ZHao+2qTQUplPKGT7XflbUjt/VoVkPqGka9z8/lPDe/Yf+4dX31n92cYtOf8398//JYedvKg9oLgRAwBQRpUqAFJfLr627pDa8+d0+X/bXr4uJP3q+K3VFs6/5ohtTynw6viGaW0Om3The9PafDfZk2PtI02OnnNx7uvfmbfC6RfLBxxbcKnstY80zX9yIZnH5srfkn4HoISz2vrM/GMLyXxC8de4z1/vPeeMyfVG3NP1vNqHf3/f5CkFjt3Ji9oDih0xAABlU+kCILX57ZEN2yz71bxR/1Nx23faJHewq2d1Pn147ZkLbzh2n3VPdPrJ6PqL53bJv8PN+Xz8c1e98kiz9EeHvprf5Zh+P1r4bPdjt76NsP73lx/Xct2I3E/AZ94rp9bNuqTuqFMXZ3zUPvuTmZfUaPfFvblPLqTAsTm2bSnprNJZMqXl+1Na7Z882JFtSyUPil0t/UGpsedlzdz2SaRPJ15w5IwOecfu3EXtAcWPGACAMqmUAZAkwHszrmk9+NMLet7Q5sxa/+972etWvrZ46t3973237fQne9VLv6q/cfnIxo0n1hhwd/dmJxySWrV05vCufd9oN3v2Ncclr7Qnt7sfT21X65b9hkwe2qr2wV+tfG5c36Fjn3+22uDc78ApcJ+d2vTW/c0aja+e82U7Vcp//OaC39w84Dcvrjjlvh3+wYGCx6ZlbCnhrDYuvrb6DQeMX3jbyfuvWnhzoy6bRizPeWcgX8ZSieJW2/K3kefWf6TmnaN7Nj3hsP0+X/mnxwdddd34ag+t3Xrszl3UHlDSiAEAKINKHQCJ7HXLpt51x32TF7z8tzUb9698TN1z2lzX+8YLqud/qj3707882qfnsMnPZn2cqlDt9DbdBw/seHz+d+SntvxzwaCrbxozf9lH2RVrnN35tiE1xp3QYWqq1dQPW0+pkvN9/PmfmE/eH3h9Qt8egx97esXqLQf9qH7bm++oO/P0S6clT857wlbJDXrud/nnH1t4S/FntTFr3JUde0995YMN3z3ipxcNHXd3q6O3vkORKLxUsqm41Tb9Y97gG/o8OG/pPz7ft3L1Bh26nby8S5/ZGef8dS9qz/gaIwYAoKzZlQBg72bEAAABCYC4jBgAICABEJcRAwAEJADiMmIAgIAEQFxGDAAQkACIy4gBAAISAHEZMQBAQKUNgA0F/h5W+uFBrdJf4f9v+QL7b1hysZnXvlOSY/9j/keVMGIAAMqi3RQAie23lFW7eKW7ePjuU8KIAQAoiwTAztvFK93Fw3efEkYMAEBZVKoASG5hD2o19cvkX7nKV7587nsjP73o0CnNX+i0okefsU+9ubpc5VrnXTl0ZK+zq5RPnpFKZa9/fUK/nsMmPbNiTarCsfVbdxsysNPx3yuX7MmUu3Kq1biX2q3od/v4Rcv+8cmXm/ZrmfOBmeJWWL9sQr8eQx57esXqLQf9qH7bXrf/ZObpl05LtZr6YespVdIL5qyQPHHbj8jbUuSy2Wv+OLpP/zGzXsj68ItvVzrmpPPbdr7q8hZ1lnRMDi907Q+c91URiyTyz21NquJxZ11225Aaj57yhAAAAOAbUaoASCS30YVexk62VOqZddopl93Uv+OpR+639rUpN7W7eeOg1ye2rFAutSlrVOPzx/6w78jeLeodWn7NsifvubHfGx3mTO98VG4eZEjWqdj1tdOb9Rp4S4s6B38yv9s5bT4auHpK8/8rcoVNb41p1mhctf4jbmxSp0q5NW8sGNOr/4N/evvUUWtyTi9ZcPtTzdtS5IkdufI3jc+bftrIEVecdsxB2R8vnz+y6xWDnjtjcpINqW2H5ypykcLntvr1J++7ecSMFUtrDV297fBvSgkjBgCgLNqtAVDh+h9MW/7Azw9MHiU2vzXs9LP/OuDt+8/dd90TF9cZUXfBgq7H7JPsSMteNbFNwxevWTKiQf4CWyXrVOxZ9clld52WuaeYFU6Yd2m90acsmn1F1byYyF47o1PNDhvu/Tjn9JIFtz/VrVuKXnbI2s5Vx5z15pzLKuW+jp/a8vcRZ5zwXNdV2wdA0Yvs6Nw+mtSmRufU/bnn9o0qYcQAAJRFuzUADn2k8d9ndTooeZDj80ktqk5v98HkluWfvq7mrdUWzr/miLzb4ORGeNWDTRqtHPT8gDrfSh5lSK8zofnK6e23dkSOjUWv8MzZvz3+1mqL5l2df4+d7Fr7SJOj51zyYc7ppRfc/lRzthRzYksuXtLopEGbOva9+crW/33cwZknmX947oI7e24fjW147B86557bN6qEEQMAUBbt3gAoYktqRodKF05cX3i5cgdfPPODhxvnPz9X/lGZ2zcUvcI7zR875tEm7zze9oBkQ57MRTL/nSt/S7En1uizl8YOvO2BWUuWrf7Of53wszOb/+JXXc4/+jsZh+cuuCvn9o0qYcQAAJRF/5YAKP9M19p31XthxrY3B4qRf1T+OomNRa+Q7Ko54NhCL8Bnvsq+YVqbwyZd+N60Nt9N9uTIeX/g4mTv1zuxDatXvPDU7MdG3vVE5cEvTutYdWPBM0xOoKhFkl3bn9vaR5rm/vTcw785JYwYAICyqPQB0OawKS3fn9Jq/+RBju1v3Ldt+XR25zOfaLHogYbfTzbn+mJJt5OuPXDCi/139BGg3KOSB/mKWeGGdy+vO+rUxXMyPme/Zkrb4y7NHp37Ofuv5nc5pt+PFj7b/ditH9Ff//vLj2u5bkR6b9HLPlt70OH3nbHsqfxP72Sv/m3jWou6vPto01Shay96kR2d2yczL6nR7outv5/wjSphxAAAlEWlDYCNi6+tfsMB4xfedvL+qxbe3KjLphFLG9x3RMEb94xb+U1/Hdu63cwT+9x22Vk1Ku+zNmvxw72vHfmtW5+f1O7wvDvjPBlHZSpmhS1v3d+s0fjqA3K/aWfNsvmje985ddmrtYflftNO9sdT29W6Zb8hk4e2qn3wVyufG9d36Njnn602OGdvkcte+Fzbg9u9cN7gsUMvOe2Yg8t98taikVdf80Kbp2b/4oebC1/7nXPrjOy4o0UKnVv5j99c8JubB/zmxRWn3Jf7DUXfqBJGDABAWVTaAEhtzBp3ZcfeU1/5YMN3j/jpRYPafdj56mlfplL7532/fnIfn/t9+Xlbvsj63aDeg8cveO39z/c9pGaDtt2HDGhfM/Oz8Wn5RyXyDsxXzArrX5/Qt8fgx55esTq7QvUzL7lt+I8nnrrtu/a3/HPBoKtvGjN/2UfZFWuc3fm2ITXGndAh/08BFLHshvefm3jv3Q/NfuGNv76/ft9Dqp/c9Mr+A688uWK5wtc+dNzdrY7+aseLpG07t5y/UXDzHXW3/o2CnJ/+DSppxAAAlEGlDoD/bElI7OhtBDLt3SMGAKBUBEBce/eIAQAolbIYAMndf+7niLb7EBGZ9uIRAwBQWmUxAPh6jBgAICABEJcRAwAEJADiMmIAgIAEQFxGDAAQkACIy4gBAAISAHEZMQBAQKUKgNzv2Uy1fOjZpq/0vmPSkrc/KVep+hntegwZ0KHWgcn+VOqLd+aNGjj4t0/+6W//+s5RP7vg+iGDrzjxoHLJjlzrl03o12NI/h/H7XX7TzL+OG72+tcn9Os5bNIzK9akKhxbv3W3IQM7Hf+9bQcXqYgDt55tq3EvtVvR7/bxi5b945MvN+3XcuqHradU2dH2nHP49LXx/XoNn5xeqmL1Bm263THgoh/nnENRq6WPSnbvTYobMQAAZVSpAiCR3AdX/OVzP6p3Wd87rjqvZsUtq5bOGn79gLc7zZ3R5UffytnbfZ8R0+5qf/zBny+fcUvH3hsGvvxQkx8kB6ZSm94a06zRuGr9R9zYpE6VcmveWDCmV/8H//T2qaPWpO+hN2WNanz+2B/2Hdm7Rb1Dy69Z9uQ9N/Z7o8Oc6Z2PKp8cW4ziDkyfT9fXTm/Wa+AtLeoc/Mn8bue0+Wjg6iktUzve3uyd+xo3nlhzQPoMD8lOX1rXfm92mDP76urfTn5QkasJAAAA/uPtQgBUuGa/8W+Ou+Dg9OviaVtWjml01h9/+eeHmnw/eZQhe/Vvm5z44vVZ95+zb/Jo3axL640+ZdHsK6rm3dFnr53RqWaHDfd+nNxDr3vi4joj6i5Y0PWYfZI9admrJrZp+OI1S0Y02G/z63eefc6w1E0Lnrqhdt7uPMUemJxtxZ5Vn1x212kFb9J3vP1fMzrVHXPa03O2nWFyaQ3PXHLly+Oapxtmx0ftfUoYMQAAZdEuBECVhxv9fdbF+ff/6dvkkeee9Va/N+5ukPNC+TbJkw+b2vqDyenXyDc+07XmrdUWzbs6/+46uVVf+0iTo+dc8uGUluWfvi7Zu3D+NUdk7F31YJNGKwc9P6BO6rXhZ509NHXTU4u6/TjvNn+rjcUeuHla60MnNF85vX3uB5TyJSe2/fb0UgOrFzzDnEvL6vvGPWckl7bDo/ZCJYwYAICyaBcC4NAprf+ZvGafPMjz2aQWR83q+N6EJmueGt6z/9g/vPrO6s82bslZZf/cz/cnB87oUPXRJu883vaAZGue/NVSMzpUunDi+pxDMpQ7+OKZHzzcODm8KMmyxRyYylu/0Ar5Pzdz+4bp7auOb1roDPMurXnyzB0etRcqYcQAAJRFuxAAVcae/9bczpULvgNw9lv9Xrtu6c/PmFxvxD1dz6t9+Pf3TV5GT56cf8ecfgdgwLGFXqr/aGzDY//QOf0OwDNda99V74UZnQ5KNu+MZNliDsw8gUw73J5+B6DwmwnpSyvwDsB2R+2FShgxAABl0S4EQIWryj305qTWlfIKYEvO7wBc/fL9m686Yux5WTO33Yt/OvGCI2d02HrHvG7WJXVHnbo44xP22WumtD3u0uzR6d8B+HR25zOfaLHogYbbfo/giyXdTrr2wAkv9q/zreRRUYo9MP0RoB3dsidXsYPt657o9JPR9RfP7ZJfAMmlNTzzuateeaRZevUdH7X3KWHEAACURbsQAJVuer1OjZY33P7L82tVzP5w6czhXftndZr7RJf/enfkufUfqXnn6J5NTzhsv89X/unxQVddN77aQ2u33jFveuv+Zo3GVx+Q/o6dKuXWLJs/uvedU5e9WntYzhfpbPrr2NbtZp7Y57bLzqpReZ+1WYsf7n3tyG/d+vykdoeXL+6XgIs9MDnbHd6yF7F94/KRjRtPrDHg7u7NTjgktSp9aX3faDd79jXHJa//F3nUXqeEEQMAUBbtQgAcOqXZc+1f797n4cU5X5Z/RrueQ2/N/TsAm/4xb/ANfR6ct/Qfn+9buXqDDt1OXt6lz+ytX/Of7E+tf31C3x6D038HILtC9TMvGTi01oQGc9q/P+nCnL1fZP1uUO/B4xe89v7n+x5Ss0Hb7kMGtK+Z/kD+pqJ/CTjHjg9MTvWgVlO/TPZn/CpCoqjtadmf/uXRPj2HTX426+NUhWqnt+k+eGDH4zP+DsCOj9rLlDBiAADKol0KgN31Kvjm5XfUv/DjEUuHnZzzCjv/HiWMGACAsuibCIDk2Io3bB4y6dcdT6p6wOY1K54adX3XBefMWHhTbff//04ljBgAgLKoVAGQ3MHnfgymlJ+B+fLdRQ8OGf7wvD+vePeT8pWqndy0S/+BV51SKe9Xbvn3KG7EAACUUaUKAMoEIwYACEgAxGXEAAABCYC4jBgAIKBvNAA2Ptezycymswaf6pd/vwn/jhEDAPAfprQBsGEXvgUo38YlPRrNbDpnSP3dHgC75fT2CrtwpSWMGACAsugbDYA95z/89HajXbjSEkYMAEBZJAD2crtwpSWMGACAsqhUAZDcdOb+HYCtyle+tFeDxwY9nmo17qV2K/rdPn7Rsn988uWm/VpO/vNPf31C9z9uTKW+fdIdf3m+xyGPNj780rkbkocnD3v1Vy/WazM1lfeXBHLXTLWasLTTih59xj715upylWudd+XQkb3OrrL1DwSsXzahX48hjz29YvWWg35Uv22v238y8/RLp+WvkC93qczTu3zuew+ct+G18f16DZ/8zIo1qYrVG7TpdseAi378vXLJ/h3IXvfqw717DJ/63N8/P/CIGj/9+WU3drukQdWcH5L9aVHr5P7coi6h+L1p2etfn9Cv57BJ6ZUrHFu/dbchAzsdn7NyUeeTu2bhKz1/3+SfX0NxIwYAoIwqVQAkklvPQi88J1sqdn3t9Ga9Bt7Sos7Bn8zvdk6bjwauntJ8eb+Trzlw4uLu1fZJnpRKffVSrxN/sc9DL91W79vpQzIXSR5W6pl12imX3dS/46lH7rf2tcnd2vbefPvrE1tWKJfa9NaYZo3GVes/4sYmdaqUW/PGgjG9+j/4p7dPHbUm7/BMyVKZK6dSG1fc17jxxJoD0ocfkr1q6azhXfu92WHO7Kur7+D3Dza+eU+TDv/b+N67Ljmp8sb/W3TvLy8e8L9nTM7JjBLWSX5u0ZdQ/N5NWaManz/2h31H9m5R79Dya5Y9ec+N/d7oMGd656PKF3M+6TULXunXV8KIAQAoi3ZvAPSs+uSyu07L35Rj8/LBp7f99N6XBiV3/KnUp3Mvr3fPiU/NvfKI5FXv5JDMRZKHFa7/wbTlD/z8wORRYnPW0NPO+dutb99/7r7rZl1ab/Qpi2ZfUTU5Lkf22hmdanbYcO/HeYdnSpbKXDn1rxmd6o457ek52w7fsnJMwzOXXPnyuOY/SB4V8K/pHX82penzE1sdnNy0JzavGHzKSS/f9FGyWknrJD+3yEsofu+6Jy6uM6LuggVdj8kNpeQCV01s0/DFa5aMqDO3yPPJWbPAle6EEkYMAEBZtFsD4NAJzVdOb7/15jbflr/fc87/rLjl1fvO+k72Px+54MzFnV98qMn3kx05h2Qskn74SOO/z+p0UPIgx+eTWlSd3u6DyS3LP9O15q3VFs27Ov/GO7lBXvtIk6PnXPJh3uGZ0ktlrLzx6etqDqxe8PAtK0eee1ZW3zfuOSMdJhnST7691jNPdjks9347Q3pXseukf24Rl5CcSTF7yycr31pt4fxr0mG0VfaqB5s0WjnombPGHl/E+STSa2Zc6c4oYcQAAJRFuzcACm7Jk/3+A41OXXz5q+Obrbrzv9t98us/pj/+k1bokEIPE/lbUjM6VH20yTuPtz0g2Zonf2/+8/MV2rVhevuq45sWOvyzSS2OmtXxvQnNc5+TL/3kCc3+b1qb7yYPCkrvKnadQj83kbkl89+58rckF1jpwonrC/8fL3fwxTPfaTbxmCLOJ5G/Qv6aX1sJIwYAoCz6dwRAKnv1hBYnPNZ4yc0rLrz1qOlzrzpy66vchQ4p9DCRvyX9DsCAYwu9QP7R2IbH/qHz130HoPDr6wVeuc+0cfG1Ne6o9eyTV27/inuJ6xT6uYnMLZn/zpW/JbnA2nfVe2HGtjcH8hVzPon8FfLX/NpKGDEAAGVR6QOgzWFTWr4/pdX+yYMcxd6J/ut3HWre9mG1TUfe8Oxvm6Q/LJ+j0CGFHia2bVk365K6o05dnPHh++w1U9oed2n26CJ+B6Dg6a17otNPRtdfPLdL/p17zmf3n7vqlUea5XwaKUP2J9M6/PTx5n+c2LpC7h33pr8MOLn9F6NfHXzSt0paZ9sJJw9yZG7J/HeubVs+nd35zCdaLHqgYXqZXF8s6XbStQdOeOH6ty4u6nzSKxS80p1QwogBACiLShsAycvS1W84YPzC207ef9XCmxt12TRiaYP7jsi9l012b+ez319+bPM/XrrklUEnJi+Ub7Xt9jd5sN3DRMaWTW/d36zR+OoD0l+/U6XcmmXzR/e+c+qyV2sPW53eW9h2pzf8iWpDm0+sMeDu7s1OOCS1aunM4V37vtFu9uxrjtt2Ovk2LB1yfqc/t7zvzkt+dsiGt+YO/sW1L7ZYsPCG4/ZJFl4+snHjotfJOOGtMrdk/jtXxpZNfx3but3ME/vcdtlZNSrvszZr8cO9rx35rVufn9Tu8I1Fn8/2Vzpiec57EV9DCSMGAKAsKm0ApDZmjbuyY++pr3yw4btH/PSiQe0+7Hz1tNwvpN9/uy/mT3yx6NoTB9f8/e/zP/6TvvfN/Q773OenCj5MDi/0hP1S61+f0LfH4PTfAciuUP3MSwYOrTWhwZz270+6MHlyYQVPb+i4u1sd9flfHu3Tc9jkZ7M+TlWodnqb7oMHdtz6Lfvby1793L09et4z/aV3N/zguHOvHDaqz/mHp2+3E9mfFrXOdidcYMuHradUyfv39ntztnyR9btBvQePX/Da+5/ve0jNBm27DxnQvmbOrxsUcz7bX+nRW/eUpKQRAwBQBpU6AHZK9prJ7Ro82f65h5rmf/xn121efkf9Cz8esXTYyV/vBW8K2a0jBgBg77CHAyB5kbvykFp/uDt147Vf3fPHzI//7LRkqYo3bB4y6dcdT6p6wOY1K54adX3XBefMWHhT7V1YNLTdM2IAAPYqez4ADmo9bUvFE69+dM6d/1M5/UGZUvvy3UUPDhn+8Lw/r3j3k/KVqp3ctEv/gVedUinvI0XsrN0zYgAA9ip7OAD4D2bEAAABCYC4jBgAICABEJcRAwAEJADiMmIAgIAEQFxGDAAQkACIy4gBAAISAHEZMQBAQAIgLiMGAAhIAMRlxAAAAQmAuIwYACAgARCXEQMABCQA4jJiAICABEBcRgwAEJAAiMuIAQACEgBxGTEAQEACIC4jBgAISADEZcQAAAEJgLiMGAAgIAEQlxEDAAQkAOIyYgCAgARAXEYMABCQAIjLiAEAAhIAcRkxAEBAAiAuIwYACEgAxGXEAAABCYC4jBgAICABEJcRAwAEJADiMmIAgIAEQFxGDAAQkACIy4gBAAISAHEZMQBAQAIgLiMGAAhIAMRlxAAAAQmAuIwYACAgARCXEQMABCQA4jJiAICABEBcRgwAEJAAiMuIAQACEgBxGTEAQEACIC4jBgAISADEZcQAAAEJgLiMGAAgIAEQlxEDAAQkAOIyYgCAgARAXEYMABCQAIjLiAEAAhIAcRkxAEBAAiAuIwYACEgAxGXEAAABCYC4jBgAICABEJcRAwAEJADiMmIAgIAEQFxGDAAQkACIy4gBAAISAHEZMQBAQAIgLiMGAAhIAMRlxAAAAQmAuIwYACAgARCXEQMABCQA4jJiAICABEBcRgwAEJAAiMuIAQACEgBxGTEAQEACIC4jBgAISADEZcQAAAEJgLiMGAAgIAEQlxEDAAQkAOIyYgCAgARAXEYMABCQAIjLiAEAAiouACjzihg+AABlVpEBAAAAlD0CAAAAAhEAAAAQiAAAAIBABAAAAAQiAAAAIBABAAAAgQgAAAAIRAAAAEAgAgAAAAIRAAAAEIgAAACAQAQAAAAEIgAAACAQAQAAAIEIAAAACEQAAABAIAIAAAACEQAAABCIAAAAgEAEAAAABCIAAAAgEAEAAACBCAAAAAhEAAAAQCACAAAAAhEAAAAQiAAAAIBABAAAAAQiAAAAIBABAAAAgQgAAAAIRAAAAEAgAgAAAAIRAAAAEIgAAACAQAQAAAAEIgAAACAQAQAAAIEIAAAACEQAAABAIAIAAAACEQAAABCIAAAAgEAEAAAABCIAAAAgEAEAAACBCAAAAAhEAAAAQCACAAAAAhEAAAAQiAAAAIBABAAAAAQiAAAAIBABAAAAgQgAAAAIRAAAAEAgAgAAAAIRAAAAEIgAAACAQAQAAAAEIgAAACAQAQAAAIEIAAAACEQAAABAIAIAAAACEQAAABCIAAAAgEAEAAAABCIAAAAgEAEAAACBCAAAAAhEAAAAQCACAAAAAhEAAAAQiAAAAIBABAAAAAQiAAAAIBABAAAAgQgAAAAIRAAAAEAgAgAAAAIRAAAAEIgAAACAQAQAAAAEIgAAACAQAQAAAIEIAAAACEQAAABAIAIAAAACEQAAABCIAAAAgEAEAAAABCIAAAAgEAEAAACBCAAAAAhEAAAAQCACAAAAAhEAAAAQiAAAAIBABAAAAAQiAAAAIBABAAAAgQgAAAAIRAAAAEAgAgAAAAIRAAAAEIgAAACAQAQAAAAEIgAAACAQAQAAAIEIAAAACEQAAABAIAIAAAACEQAAABCIAAAAgEAEAAAABCIAAAAgEAEAAACBCAAAAAhEAAAAQCACAAAAAhEAAAAQiAAAAIBABAAAAAQiAAAAIBABAAAAgQgAAAAIRAAAAEAgAgAAAAIRAAAAEIgAAACAQAQAAAAEIgAAACAQAQAAAIEIAAAACEQAAABAIAIAAAACEQAAABCIAAAAgEAEAAAABCIAAAAgEAEAAACBCAAAAAhEAAAAQCACAAAAAhEAAAAQiAAAAIBABAAAAAQiAAAAIBABAAAAgQgAAAAIRAAAAEAgAgAAAAIRAAAAEIgAAACAQAQAAAAEIgAAACAQAQAAAIEIAAAACEQAAABAIAIAAAACEQAAABCIAAAAgEAEAAAABCIAAAAgEAEAAACBCAAAAAhEAAAAQCACAAAAAhEAAAAQiAAAAIBABAAAAAQiAAAAIBABAAAAgQgAAAAIRAAAAEAgAgAAAAIRAAAAEIgAAACAQAQAAAAEIgAAACAQAQAAAIEIAAAACEQAAABAIAIAAAACEQAAABCIAAAAgEAEAAAABCIAAAAgEAEAAACBCAAAAAhEAAAAQCACAAAAAhEAAAAQiAAAAIBABAAAAAQiAAAAIBABAAAAgQgAAAAIRAAAAEAgAgAAAAIRAAAAEIgAAACAQAQAAAAEIgAAACAQAQAAAIEIAAAACEQAAABAIAIAAAACEQAAABCIAAAAgEAEAAAABCIAAAAgEAEAAACBCAAAAAhEAAAAQCACAAAAAhEAAAAQiAAAAIBABAAAAAQiAAAAIBABAAAAgQgAAAAIRAAAAEAgAgAAAAIRAAAAEIgAAACAQAQAAAAEIgAAACAQAQAAAIEIAAAACEQAAABAIAIAAAACEQAAABCIAAAAgEAEAAAABCIAAAAgEAEAAACBCAAAAAhEAAAAQCACAAAAAhEAAAAQiAAAAIBABAAAAAQiAAAAIBABAAAAgQgAAAAIRAAAAEAgAgAAAAIRAAAAEIgAAACAQAQAAAAEIgAAACAQAQAAAIEIAAAACEQAAABAIAIAAAACEQAAABCIAAAAgEAEAAAABCIAAAAgEAEAAACBCAAAAAhEAAAAQCACAAAAAhEAAAAQiAAAAIBABAAAAAQiAAAAIBABAAAAgQgAAAAIRAAAAEAgAgAAAAIRAAAAEIgAAACAQAQAAAAEIgAAACAQAQAAAIEIAAAACEQAAABAIAIAAAACEQAAABCIAAAAgEAEAAAABCIAAAAgEAEAAACBCAAAAAhEAAAAQCACAAAAAhEAAAAQiAAAAIBABAAAAAQiAAAAIBABAAAAgQgAAAAIRAAAAEAgAgAAAAIRAAAAEIgAAACAQAQAAAAEIgAAACAQAQAAAIEIAAAACEQAAABAIAIAAAACEQAAABCIAAAAgEAEAAAABCIAAAAgEAEAAACBCAAAAAhEAAAAQCACAAAAAhEAAAAQiAAAAIBABAAAAAQiAAAAIBABAAAAgQgAAAAIRAAAAEAgAgAAAAIRAAAAEIgAAACAQAQAAAAEIgAAACAQAQAAAIEIAAAACEQAAABAIAIAAAACEQAAABCIAAAAgEAEAAAABCIAAAAgEAEAAACBCAAAAAhEAAAAQCACAAAAAhEAAAAQiAAAAIBABAAAAAQiAAAAIBABAAAAgQgAAAAIRAAAAEAgAgAAAAIRAAAAEIgAAACAQAQAAAAEIgAAACAQAQAAAIEIAAAACEQAAABAIAIAAAACEQAAABCIAAAAgEAEAAAABCIAAAAgEAEAAACBCAAAAAhEAAAAQCACAAAAAhEAAAAQiAAAAIBABAAAAAQiAAAAIBABAAAAgQgAAAAIRAAAAEAgAgAAAAIRAAAAEIgAAACAQAQAAAAEIgAAACAQAQAAAIEIAAAACEQAAABAIAIAAAACEQAAABCIAAAAgEAEAAAABCIAAAAgEAEAAACBCAAAAAhEAAAAQCACAAAAAhEAAAAQiAAAAIBABAAAAAQiAAAAIBABAAAAgQgAAAAIRAAAAEAgAgAAAAIRAAAAEIgAAACAQAQAAAAEIgAAACAQAQAAAIEIAAAACEQAAABAIAIAAAACEQAAABCIAAAAgEAEAAAABCIAAAAgEAEAAACBCAAAAAhEAAAAQCACAAAAAhEAAAAQiAAAAIBABAAAAAQiAAAAIBABAAAAgQgAAAAIRAAAAEAgAgAAAAIRAAAAEIgAAACAQAQAAAAEIgAAACAQAQAAAIEIAAAACEQAAABAIAIAAAACEQAAABCIAAAAgEAEAAAABCIAAAAgEAEAAACBCAAAAAhEAAAAQCACAAAAAhEAAAAQiAAAAIBABAAAAAQiAAAAIBABAAAAgQgAAAAIRAAAAEAgAgAAAAIRAAAAEIgAAACAQAQAAAAEIgAAACAQAQAAAIEIAAAACEQAAABAIAIAAAACEQAAABCIAAAAgEAEAAAABCIAAAAgEAEAAACBCAAAAAhEAAAAQCACAAAAAhEAAAAQiAAAAIBABAAAAAQiAAAAIBABAAAAgQgAAAAIRAAAAEAgAgAAAAIRAAAAEIgAAACAQAQAAAAEIgAAACAQAQAAAIEIAAAACEQAAABAIAIAAAACEQAAABCIAAAAgEAEAAAABCIAAAAgEAEAAACBCAAAAAhEAAAAQCACAAAAAhEAAAAQiAAAAIBABAAAAAQiAAAAIBABAAAAgQgAAAAIRAAAAEAgAgAAAAIRAAAAEIgAAACAQAQAAAAEIgAAACAQAQAAAIEIAAAACEQAAABAIAIAAAACEQAAABCIAAAAgEAEAAAABCIAAAAgEAEAAACBCAAAAAhEAAAAQCACAAAAAhEAAAAQiAAAAIBABAAAAAQiAAAAIBABAAAAgQgAAAAIRAAAAEAgAgAAAAIRAAAAEIgAAACAQAQAAAAEIgAAACAQAQAAAIEIAAAACEQAAABAIAIAAAACEQAAABCIAAAAgEAEAAAABCIAAAAgEAEAAACBCAAAAAhEAAAAQCACAAAAAhEAAAAQiAAAAIBABAAAAAQiAAAAIBABAAAAgQgAAAAIRAAAAEAgAgAAAAIRAAAAEIgAAACAQAQAAAAEIgAAACAQAQAAAIEIAAAACEQAAABAIAIAAAACEQAAABCIAAAAgEAEAAAABCIAAAAgEAEAAACBCAAAAAhEAAAAQCACAAAAAhEAAAAQiAAAAIBABAAAAAQiAAAAIBABAAAAgQgAAAAIRAAAAEAgAgAAAAIRAAAAEIgAAACAQAQAAAAEIgAAACAQAQAAAIEIAAAACEQAAABAIAIAAAAC+f96b8PdM94nHQAAAABJRU5ErkJggg==\nE         + iVBORw0KGgoAAAANSUhEUgAABAAAAALkCAIAAAAeetB6AAAsmklEQVR4nO3de4DVc/748SnXXXZ1x36zy7Z085VlI7RlW+q3ulndS2kLhWyWEOkyElPoG0aEFql0s919rUouW9RebEtRrdtGvlK0tCpd5vc5M9M0c6amwve7Na/H4w/mc87n8zmfOa9/3s9zTmfK5OTkZAAAADEIAAAACEQAAABAIAIAAAACEQAAABCIAAAAgEAEAAAABCIAAAAgkN0GQJkyyX8o5XYzfAAASq2SAmA391BKGDEAQEACIC4jBgAIqHQFwJaFfZvPaDEz6+xDkg324IAcMQAAX0+JAbBpSttybSZvysg4vM3k9ZNaH5bcvH/bsuDGpjNazB56zgERAJtzn96MvXhu937PfZEa8a6HDwBAqVViACT3JEvPYya1/Z9vct3JTnv/9O79nnstf8QAAEQiAP6t9v7p3fs991r+iAEAiOTrB0DO56+NHXjTXRNfXL4uo2L1Bu363JF58X9+J/UlosmxuR9cGfPHDssH3j52/tIP1m/aeljr3M+x5Gx4fdzAvndOSB1V4cRz2vYZOrjLKd8pk39I68cXNF/Q65Ynl3xe6fT2maOGX7T1yd49Bk16dX25H3fMevzejicempw+Xf6xOz4nk7/Z+tGXWrza744JC/6+vkyl6g073Dg0s1PtI5Pdc9a98kD/QaNmLlrx0cZDKlWr26R99ysubVVnQefUUTtOksg/T+4tGXk/f6XLy7Vh6biBNw598oXkt65Yo1G324bWfOKs6Tue3t0/k4nkMooMYvc7J3umLnKXT3ty9075IwYAIJKvGwBblt/frNn4Wpkjrmtep0rOmiUz7+o98I1Os2ddVT33Y/jJ4RV7v/bTljcNvqVVnfLr5/Q5r93Hg9dOuvC9kc2ajP7BgOx+rU4/puy6pc/ce93AZZ1mT+1+fNnUIZVueKdlp8zM6887Zv38gRd1eP74pjVrdBh0w3nHfDqv/0Vd3+r9t2ldjk4tdYtJji18tclmxSsX/uj0bgPuuKJxrYrbU5d3Tebfuzw9rccPP3ioWeOp9bNHXF6/WrmcT96ck9378iELG05MVsnJKr/wSRLJeQpuSX7+qpe3deWolk3HnDQo9VwdXWbt68/cf/OIacuX1B62NnXmPT+TBdfwVZ/2vGML5I8YAIBIvmYA/HNal9NG1X9h9uVVyyZbKdtXjbrg3AU9/zLmwqOSreTwin2rPrN0eP3Cx382/ZI6I06bO7d3tYOSrZScNePbXbC414IRDQ7LPeS43y+7+5zU6+jbPxh5/onDT52zY3NV9vlnL7nhrYebpLaKSY4tfLXJZoVeh419Y8wvy+9YkCeX17TRK1f++cHtV1Yd1eiN2d0q5d+z/Z0RDU9d2HvNXgTAV7y8z2b+6vQHzpo/a+dzlfPxhHY1u2c8+Ely5r14Jguu4Ss+7enyRwwAQCRfLwC2vPDrWoOrz3/2qoKFaGopmn1+oxUDlt3bMHkxOnX4uAtXTe2Y+tDNDqmjbj3puTm9jtt5VM6aR5o3XTXk5cw625JDnmjx7vSLUyvZjIxNk9scO7ZlwebGSW3+46l2H07c9fWkHq7Q1SabRz/W9J2ZlxSs//Mvb+XA13u/3qzukK2dB9zcs+3PapQ/OLlnh+SowidJFL4l9fNXurwtL/ZOfuuiz1XOx6MvOPH33T+a1Lps8pzs8ZnccQ2pJ3CPOxd72ovJHzEAAJF8vQDYPLVj1bEt3n2q/RHJxg7/mtDq+Jmd3x93YXLILg/fPK1TpYvGb0jOXkSZ8pfM+PCxZmkvwKedIW0zTdq9aZt5dlxey3/9cfTg2x6euWDp2m/98NQzz73wst/0aHLCt3Z1VOFbCv+cKHmzsOS3rvpE87TnqmD/jH15Jr/a015M/ogBAIjk6wVA6qXo9Nfyi70UXezw5LXwk4efvmhal3LJRjFph5S8mSbt3mTz6NFNVj7dvXLRdwB+vnLg0nsaJJeXa/Pa5YvmzXoye/j0ylmLp3SuumVKu2MnXPT+lHbfTu7M9enjzU+YfUnyOn1y2uScaQ9RwmZhyW9dK/PEos9VzqePt8g7c+odgL1+Jr/a015M/ogBAIjkKwXAzhs/m97lxw+c8/zTPQqWorkfRl94xauPt/xusrVzz2SjwOezup87vdX8hy9I7ZJn44I+da8+ctziQbkfASp0SNoZ0jbTpN2bbFa4osyjb0xou+OT/qnLa9rolav+8uCm7lXub7h0XsGnaHLW/rZZ7fk9/vFEizJzelQb+KPnXrr+xPx/obDhvy+t0fqzEalP6ueeM+0hStgs4rOZXU8befbzhT64n7N+RteaHTbelzrzPj2T+7TzbuWPGACASL5mAGRseTO7WbPxNTPvub7lqVUy1iyZcVfvAcs6zJrVq0byQnSRPQvb+tboth1m/KT/bd0a1ax80Kcrnn+s39XZB9/68oQO3yubdkjJm2nS7k02K93wep2ara+9/comtSvmfJS6vEErujw9vcdx09qW77CocdboYV3rVytfZv3K+dlX9VrUbt6sy35Q5pPJHWrfctjQicPanFz+y1ULxwwYNvrll07KyvsWneScaQ9RwmZRW1c+2LLp2OqZqa/uObrsJ2/MfejmzIcWLz/r/nWp/ffpmdynnXcnf8QAAERSYgBsyv1G+U3JVjGH534vfmp9mfP5357o3/fOiS+t+CSjwkk/bXd91uDOpxT6Qvq8w3fun2/jit8N6Zc1du5rq784tEqtBu2vH5rZsdYRBYfk7Z9R4mahs6WkHZvcm9xyzKSWCzu+fn3/x57P/b78hh36Drs19+8AbF69cPx99zw6a9Gyt1ZvOLRK9Xoteg4a3LNexdSFb/+fuUOuumHUnKUf51Ss+fPutw2tOebUTpMz2kz+qO2kows9RNr1pG0mF1DMhtfHDbgx68kXlq/dXu5H57S/+Y7TZvz0V1OSM6f234tncueZ92LnxM79dyE14l0PHwCAUqvEANj1PQeOZCm8Fy+Eh1UaRgwAwD4SAHGVhhEDALCPSm8AJKv/vE/ClPgxmMgO+BEDALDvSm8AsCdGDAAQkACIy4gBAAISAHEZMQBAQF81ALYs7Nt8RouZWWfnfu88B6I9jBgAgNLoKwfAghubzmgxe+g5eQGwef/4vp395DIOEHsYMQAApdFXDYA0+8nKez+5jAPEvo0YAIBSQQDEtW8jBgCgVNhDAOSse+WB/oNGzVy04qONh1SqVrdJ++5XXNqqzoLO5dpMztjx/frJsjvZ3JQclKds5Uuffv/hJocmP+608d1nRw7O+u0zf3r7n986/sxfXjM06/KflCuT3JF/eEbrxxc0X9DrlieXfF7p9PaZo4ZftPXJ3j0GTXp1fbkfd8x6/N6OJxacL2fD6+MG9r1zwovL12VUOPGctn2GDu5yynfK5J1nD5fBTnkjBgAglBIDYNt7DzVrPLV+9ojL61crl/PJm3Oye18+ZGHDicm6P6Poa+3J4rvwZnHJDhWvP2jElOEdTyn/xZvTbuncb/Pgvzza/KjkrkRyb6Ub3mnZKTPz+vOOWT9/4EUdnj++ac0aHQbdcN4xn87rf1HXt3r/bVqXo3N7YeuKkc2ajP7BgOx+rU4/puy6pc/ce93AZZ1mT+1+fNnUeUq+DApJjXjXwwcAoNQqMQA2Tb+46qhGb8zuVil36Z2Rsf2dEQ1PXdh7zb4HQFE5a3/b/CeLr1nx4Hl5r88nh1fse9zvl919Tmp7+wcjzz9x+Klzdmyuyj7/7CU3vJX3Yv5n0y+pM+K0uXN7Vzso2UrJWTO+3QWLey0Y0eCw5Dz7chnBpUa86+EDAFBqlRgA294e2aTukK2dB9zcs+3PapQ/OLl5h7SldtrmHiX7Hzu57YcT8/dPNo95osW70y/Oe0dg0+Q2x45tWbC5cVKb/3iqXd7OW174da1bT3puTq/jyib35MlZ80jzpquGvJxZZ1tynn25jNhSI9718AEAKLVKDICcjJxP/jh68G0Pz1ywdO23fnjqmedeeNlvejQ54Vt5S/ZCS+20zWK2rZ53V99Bo3//13fX/mvL9txHPHzHPyFIpB1ewubmaZ0qXTR+Q+4ZCilT/pIZHz7WLO19CUqUN2IAAELZQwAU2Lx2+aJ5s57MHj69ctbiKZ2rbim61C68Ri9u+1v3nddw4ukj7u3d+OTvfffQ5LX7tP33fnPLi71PHn76omldyiV3FFN4T/YkbcQAAERQYgBsmtK2yv0Nl867qmqyZk/JWfvbZrXn9/jHEy3SXmvfPKXdsZNar57U5vBko5jN0y8+bnTjFTN2rto/H//L70/rVOjwtLPtfvPzWd3Pnd5q/sMXfDfZyLNxQZ+6Vx85bvGgOttKvAyKSo1418MHAKDU2lMAlO+wqHHW6GFd61crX2b9yvnZV/Va1G7erMt+kPYOwJbnr65+7RFjn7ut3uFrnru5aY+tI968t2HeHwlObH87+/xzHq919wN9W5x67GFfrPrTU0Ou+PXYkx79dMfhRZb4e9jc+tboth1m/KT/bd0a1ax80Kcrnn+s39XZB9/68oQO3ytb/DL+fPZ9/znn0rdG/7/UPyCmiNSIdz18AABKrRIDIGfz6oXj77vn0VmLlr21esOhVarXa9Fz0OCe9Y58qm3eN+7v/Bz/lhVjenbuN/nVDzd/+7gzLh425p42J+z4lp6UrR88m3Vt/0eeXfLBF4dWrt6gU596b/boPyvvLwkkbyYUPlvJm6nHyti44ndD+mWNnfva6i8OrVKrQfvrh2Z2rHVEckexy/iv06ae33XrQ8/fcFLhqyFX7oiT/wMAEEjJAZD8/4C27e17/l+Xz//rhVtOtv4vrlSMGACAfVOqA2DbshE9xp929+AGR+X/HQMKKw0jBgBgH5XqAKBERgwAEJAAiMuIAQACEgBxGTEAQEACIC4jBgAIqFQEQNE/FLAH+7RzqXYgjRgAgG+IAIjrQBoxAADfEAEQ14E0YgAAviF7DICcz/76WL8b75q88J0vjjyu5hm/6HZdn64NquYunnM+f23swJvumvji8nUZFas3aNfnjsyL//M7qa/cTxbZ5dpMzmgzbkmX5Tf2Hz3vjbVlKtdu3HNY9k0/P7pscn/Oulce6D9o1MxFKz7aeEilanWbtO9+xaWtzjj2sIycDa+PG9j3zgmpc1Y48Zy2fYYO7nJK7jl3dyV5j7Up2SFP2cqXPv3+w42/3OV5iu/8q5saPDnkqby/SZz7WxVcfOqWvL9DnNFmzB87LB94+9j5Sz9Yv2nrYa1Td+XtfCDbMWIAAALZQwBseePe5p3+0Oy+4V3rVt7y3vz7rrwk8w8NJ+Yuf7csv79Zs/G1Mkdc17xOlZw1S2be1XvgG51mz7qq+iHJ4alldKW+K+qf1e2GQZ3P/v5hn742sU/7fttuf3186wo57z3UrPHU+tkjLq9frVzOJ2/Oye59+ZCFqdNe+N7IZk1G/2BAdr9Wpx9Tdt3SZ+69buCyTrOndj++bAlXknqsoi/qb12x2/MU33kPtyQ/V+z92k9b3jT4llZ1yq+f0+e8dh8PXpt71wEub8QAAIRScgD8c2rnMye1eHl8m/K5r8FnbFuedVbdv9zwcbL8/ee0LqeNqv/C7Murpl7ST9m+atQF5y7o+ZcxFx6VbCXr5grXHDXlzYd/cWSyldi2Ylj9896+9e8Pnp8z/eKqoxq9MbtbpbyzZmx/Z0TDUxf2XvPIIZfXGXHa3Lm9qx2U3JqSs2Z8uwsW91owos7Tu72S3McqWK+nfDb9kt2dp0HqHYMiOxc/vOgtyc8V+1Z9Zunw+slGqZI74uT/AAAEUmIAfPnCr2vdXvvFZ3ocm7fqLmRLctfg6vOfvapg/Z+s41dln99oxYBl9zZM3gNI1s3HPN7snZldyiX35PpiQquqUzt8OLH1IW+PbFJ3yNbOA27u2fZnNcofnNyXK3XOW096bk6v43aeM2fNI82brhryYqPRp+zmShKpx9qxXk+UcJ6XM+tsK7pzIu3wROFbUj+Pu3DV1I75IVN6pEa86+EDAFBqlRgAm6Z2rDqu5XtT2n07uaGozcldY1u8+1T7I5KNHf41odXxMzu/P+7C/HXzjjV0nkK35Hzyx9GDb3t45oKla7/1w1PPPPfCy37To8kJZad1qnTR+A3p11Om/CUz3m05vtpuriRR6Mwpm3d/ng8fa5ZRdOdE2uGJwrcU/rl0SY04/UkCAKCUKzEAvnz+6pp31H7pmZ7FX3ff1avsxd4BKLpuLn5Lctva5YvmzXoye/j0ylmLn/zBnacMP33RtJ1vGhTYsvsrSaSdecuLvU/ezXkSaTsnNk9pd+yEi94vVBefPt78hNmXfJS7T/H9S4vUiHc9fAAASq0SA2D7+imdznjqwlfGt62Qt+7e+rfMeh03PvDXrLoHfza9y48fOOf5p3sUFEDuvwFYeMWrj7f8brJVfN1ccEvyGnyV+xsunVfw8aGctb9tVnt+j3+MLHvludNbzX/4gtQJ8mxc0Kfu1UeOW3TNykt2dyWpM7c7dlLr1ZPaHJ7ck/h8VvfdnGfxoDrb0nbOyPhyTo9qA3/03EvXn5j/TwY2/PelNVp/NuKT3IsvuOzk59IlNeJdDx8AgFKrxADIydi8ZGiTLn9uff/dXc+ssnnl01mXXb241dznrq2RrJS3vJndrNn4mpn3XN/y1CoZa5bMuKv3gGUdZs3qVSN5/X9X6+aCW5IAKN9hUeOs0cO61q9Wvsz6lfOzr+q1qN28WZf9xzuj23aY8ZP+t3VrVLPyQZ+ueP6xfldnH3zryxM6fG/L7q8k9f5A9WuPGPvcbfUOX/PczU17bL376TrZnXd9nrLFdh7xxqA1XWrfctjQicPanFz+y1ULxwwYNvrll07Kyvuqn4LLTn4uXfJGDABAKHsIgNSr8wvvu7HvvVP/+I/NR9U4v+edI/s3+V5q0Z3I+fxvT/Tve+fEl1Z8klHhpJ+2uz5rcOdTcr+zP1k0l8v9uv3Dc79NP1k6F7nlibP/PP6+ex6dtWjZW6s3HFqler0WPQcN7lmvYurQjSt+N6Rf1ti5r63+4tAqtRq0v35oZsdauf/QoIQr2bJiTM/O/Sa/+uHmbx93xsXDxtzT5oQvd3ueXexc5n/mDrnqhlFzln6cU7Hmz7vfNrTmmFM7Tc5oM/mjtpOOzr3sROrKc3+X0mLHiAEACGSPAUCpZcQAAAEJgLiMGAAgIAEQlxEDAAQkAOIyYgCAgARAXEYMABCQAIjLiAEAAvrfDIAtC/s2n9FiZtbZuX8YYA/2aedd+vpn+Pr2h2vYa9/AiAEAONB8owGQ92X/GTu+L3/Lghubzmgxe+g5e7Ma3qedd+nrn2Ffpf2+ibRrSHbYj/+I2FcZMQAAB7hvNAAS+/eS95tX8u9b8r3/bl9xxAAAHMgEwNdT8u9b8r3/bl9xxAAAHMj2GAAb33125OCs3z7zp7f/+a3jz/zlNUOzLv9JuTLJHXk2LB038MahT76wfF1GxRqNut02tOYTZ01PLXkzin48JlkKpzZbP/pSi1f73TFhwd/Xl6lUvWGHG4dmdqp95I57d+yckZGz7pUH+g8aNXPRio82HlKpWt0m7btfcWmrM449LCPns78+1u/GuyYvfOeLI4+recYvul3Xp2uDqoelnWHH5rglXZbf2H/0vDfWlqlcu3HPYdk3/fzossn5EwVXvnZ7uR+d0/6m238846e/mlJwhiJyNrw+bmDfOye8mPyaFU48p22foYO7nPKd1LOQPFDBEj/5OfdB88+Qt7kp2SlP2cqXPv3+w00OTX7cP+wYMQAAgewpAJJVbMXrDxoxZXjHU8p/8ea0Wzr32zz4L482PyrZIyNj68pRLZuOOWnQiOua1zm6zNrXn7n/5hHTli+pPWztjhVwweI4kWxWvHLhj07vNuCOKxrXqrh9zZKZd12T+fcuT0/r8aODU/cW7Lz9vYeaNZ5aP3vE5fWrlcv55M052b0vH7Kw4cT1k1q+fW/zTn9odt/wrnUrb3lv/n1XXpL5h9TtqaMKnyGRbFbqu6L+Wd1uGNT57O8f9ulrE/u077ft9tfHt65QJu3K1y2bO+qmQY/86e9nj1y34/BCtq4Y2azJ6B8MyO7X6vRjyq5b+sy91w1c1mn21O7Hl009StqDlrC5n8kfMQAAkewpAIrKWfvb5j9ZfM2KB89LvY792cxfnf7AWfNnXV41/0X1jJyPJ7Sr2T3jwU9yl7xpy99ks0Kvw8a+MeaX5VOvnadsXzWqaaNXrvzzo82/m9xbsPPm6RdXHdXojdndKuXvuP2dEQ1PXdh7zcMHXXrmpBYvj2+Tf4Zty7POqvuXGz7OO6rQGRLJZoVrjpry5sO/ODLZSmxbMaz+eW/f+vcHzz90F1f+6bQutTptvi/vyov4bPoldUacNndu72oHJVspOWvGt7tgca8FIxqk3nlIe9ASNvczuxwxAACl274FQGpFe+zkth9OTK1ot7zYu9atJ81/9qqCVXSyNP549AUn/r77R7lL3mTnwsvfZPPox5q+M/OS/NV7yvZV2ec3Wjlw2T0NthfaefvbI5vUHbK184Cbe7b9WY3yByc75trywq9r3V77xWd6HLvzDDsl5y84QyK1+Xizd2Z2KZds5PpiQquqUzskF192V1f+6ePNT5jdNe/KC0s96K0nPTen13GFdl7zSPOmq4a8nFlnW/IoaQ+6+839zK5HDABAqbaHANi2et5dfQeN/v1f3137ry3bc/c8vOAz7tM6VX2i+btPtT8iuXWHwkvewj8n0jbz/GtCq+Nndn5/3IUZRe7N+eSPowff9vDMBUvXfuuHp5557oWX/aZHkxPKTu1YdVzL96a0+3aySzFp50/bTBTckrGnKy8s+TUrXTR+Q+7vXkiZ8pfM+PCxZkUvO/0kaZv7mbwRAwAQSokBsO2t+85rOPH0Eff2bnzy9757aPIKeOEVbeodgMwT014a//TxFifMviTvdfTCOyeSzaNHN1n5dPfKO1+/T70D8POVA5cWfQegkM1rly+aN+vJ7OHTK2ctfvK4obXvqP3SMz339h2AQpuJgltS7wAUu/LC710UlvyaJw8/fdG0ne8kFFZwzryjSt7cz6RGvOvhAwBQapUYAJumX3zc6MYrZuxc+34+/pffn9Ypf0X72cyup408+/nZhT5Jv35G15odNuZ/kj5t+ZtsVriizKNvTGi746P9yfo/9W8ArvrLo82+k9xbsHPyc5X7Gy6dV/ARnZy1v21We36P9+77stsZT134yvi2FfLOsPVvmfU6bnzgr1l1D849quAMibTNxM5bdnHl6ya1r/GrnAfyrryIz2d1P3d6q/kPX/DdZCPPxgV96l595LjFg/b4EaB2x05qvXpSm8OTjf1OasS7Hj4AAKVWiQGw7e3s8895vNbdD/Rtceqxh32x6k9PDbni12NPevTT/AXu1pUPtmw6tnpm7nfplP3kjbkP3Zz50OLlZ92f9106xVbDbSvd8Hqdmq2vvf3KJrUr5ny0ZMZdvQet6PL09PRvAUp+Lt9hUeOs0cO61q9Wvsz6lfOzr+q1qN28WZcd89rQJl3+3Pr+u7ueWWXzyqezLrt6cau5z11bI/XPc5OjCs6QSNtMFLqlyJWXWbd0zgP97p689K8n35n3/UVFbX1rdNsOM37S/7ZujWpWPujTFc8/1u/q7INvfXlCh++VLXTOlLTNLc9fXf3aI8Y+d1u9w9c8d3PTHltHLGlwf+VCXxX6b5Ua8a6HDwBAqVViAORkbP3g2axr+z/y7JIPvji0cvUGnfrUe7NH/1mF1q8bXh834Masgm/Tv/mO0/K/Tf+jtpOOzv0W/J3/ZiC1OG65sOPr1/d/7Pnl6zIqVm/Yoe+wW3f+HYCdO29evXD8ffc8OmvRsrdWbzi0SvV6LXoOGtyzXsXkdf+ctQvvu7HvvVP/+I/NR9U4v+edI/s3+V6y/E87Q0bRzbxHL3rLzivPqVD93K6Dh9Ue12B2x9UTLkp2Lmbjit8N6Zc1du5rq784tEqtBu2vH5rZsdYRaecs/qAZW1aM6dm53+RXP9z87ePOuHjYmHta/LlDsk/y/OTv8O+UN2IAAELZQwB8k5LlcuFXx/cz296845yLPhmx5M56hyRbIXzzIwYAYL8XNQCSi6l47bahE/6rc92qR2xbt3zeyGt6zz1v2nM3nBxm/f+/MGIAAPZ7/1cBkCy4y6V9PObfa9M/5j8y9K7Hnv3z8n+sL1vppHotegwafMVZlXb8o+AIvuERAwBwIPi/CgD2P0YMABCQAIjLiAEAAhIAcRkxAEBAAiAuIwYACEgAxGXEAAABCYC4jBgAICABEJcRAwAEJADiMmIAgIAEQFxGDAAQkACIy4gBAAISAHEZMQBAQAIgLiMGAAhIAMRlxAAAAQmAuIwYACAgARCXEQMABCQA4jJiAICABEBcRgwAEJAAiMuIAQACEgBxGTEAQEACIC4jBgAISADEZcQAAAEJgLiMGAAgIAEQlxEDAAQkAOIyYgCAgARAXEYMABCQAIjLiAEAAhIAcRkxAEBAAiAuIwYACEgAxGXEAAABCYC4jBgAICABEJcRAwAEJADiMmIAgIAEQFxGDAAQkACIy4gBAAISAHEZMQBAQAIgLiMGAAhIAMRlxAAAAQmAuIwYACAgARCXEQMABCQA4jJiAICABEBcRgwAEJAAiMuIAQACEgBxGTEAQEACIC4jBgAISADEZcQAAAEJgLiMGAAgIAEQlxEDAAQkAOIyYgCAgARAXEYMABCQAIjLiAEAAhIAcRkxAEBAAiAuIwYACEgAxGXEAAABCYC4jBgAICABEJcRAwAEJADiMmIAgIAEQFxGDAAQkACIy4gBAAISAHEZMQBAQAIgLiMGAAhIAMRlxAAAAQmAuIwYACAgARCXEQMABCQA4jJiAICABEBcRgwAEJAAiMuIAQACEgBxGTEAQEACIC4jBgAISADEZcQAAAEJgLiMGAAgIAEQlxEDAAQkAOIyYgCAgARAXEYMABCQAIjLiAEAAhIAcRkxAEBAAiAuIwYACEgAxGXEAAABCYC4jBgAICABEJcRAwAEJADiMmIAgIAEQFxGDAAQkACIy4gBAAISAHEZMQBAQAIgLiMGAAhIAMRlxAAAAQmAuIwYACCgkgKAUm83wwcAoNTabQAAAACljwAAAIBABAAAAAQiAAAAIBABAAAAgQgAAAAIRAAAAEAgAgAAAAIRAAAAEIgAAACAQAQAAAAEIgAAACAQAQAAAIEIAAAACEQAAABAIAIAAAACEQAAABCIAAAAgEAEAAAABCIAAAAgEAEAAACBCAAAAAhEAAAAQCACAAAAAhEAAAAQiAAAAIBABAAAAAQiAAAAIBABAAAAgQgAAAAIRAAAAEAgAgAAAAIRAAAAEIgAAACAQAQAAAAEIgAAACAQAQAAAIEIAAAACEQAAABAIAIAAAACEQAAABCIAAAAgEAEAAAABCIAAAAgEAEAAACBCAAAAAhEAAAAQCACAAAAAhEAAAAQiAAAAIBABAAAAAQiAAAAIBABAAAAgQgAAAAIRAAAAEAgAgAAAAIRAAAAEIgAAACAQAQAAAAEIgAAACAQAQAAAIEIAAAACEQAAABAIAIAAAACEQAAABCIAAAAgEAEAAAABCIAAAAgEAEAAACBCAAAAAhEAAAAQCACAAAAAhEAAAAQiAAAAIBABAAAAAQiAAAAIBABAAAAgQgAAAAIRAAAAEAgAgAAAAIRAAAAEIgAAACAQAQAAAAEIgAAACAQAQAAAIEIAAAACEQAAABAIAIAAAACEQAAABCIAAAAgEAEAAAABCIAAAAgEAEAAACBCAAAAAhEAAAAQCACAAAAAhEAAAAQiAAAAIBABAAAAAQiAAAAIBABAAAAgQgAAAAIRAAAAEAgAgAAAAIRAAAAEIgAAACAQAQAAAAEIgAAACAQAQAAAIEIAAAACEQAAABAIAIAAAACEQAAABCIAAAAgEAEAAAABCIAAAAgEAEAAACBCAAAAAhEAAAAQCACAAAAAhEAAAAQiAAAAIBABAAAAAQiAAAAIBABAAAAgQgAAAAIRAAAAEAgAgAAAAIRAAAAEIgAAACAQAQAAAAEIgAAACAQAQAAAIEIAAAACEQAAABAIAIAAAACEQAAABCIAAAAgEAEAAAABCIAAAAgEAEAAACBCAAAAAhEAAAAQCACAAAAAhEAAAAQiAAAAIBABAAAAAQiAAAAIBABAAAAgQgAAAAIRAAAAEAgAgAAAAIRAAAAEIgAAACAQAQAAAAEIgAAACAQAQAAAIEIAAAACEQAAABAIAIAAAACEQAAABCIAAAAgEAEAAAABCIAAAAgEAEAAACBCAAAAAhEAAAAQCACAAAAAhEAAAAQiAAAAIBABAAAAAQiAAAAIBABAAAAgQgAAAAIRAAAAEAgAgAAAAIRAAAAEIgAAACAQAQAAAAEIgAAACAQAQAAAIEIAAAACEQAAABAIAIAAAACEQAAABCIAAAAgEAEAAAABCIAAAAgEAEAAACBCAAAAAhEAAAAQCACAAAAAhEAAAAQiAAAAIBABAAAAAQiAAAAIBABAAAAgQgAAAAIRAAAAEAgAgAAAAIRAAAAEIgAAACAQAQAAAAEIgAAACAQAQAAAIEIAAAACEQAAABAIAIAAAACEQAAABCIAAAAgEAEAAAABCIAAAAgEAEAAACBCAAAAAhEAAAAQCACAAAAAhEAAAAQiAAAAIBABAAAAAQiAAAAIBABAAAAgQgAAAAIRAAAAEAgAgAAAAIRAAAAEIgAAACAQAQAAAAEIgAAACAQAQAAAIEIAAAACEQAAABAIAIAAAACEQAAABCIAAAAgEAEAAAABCIAAAAgEAEAAACBCAAAAAhEAAAAQCACAAAAAhEAAAAQiAAAAIBABAAAAAQiAAAAIBABAAAAgQgAAAAIRAAAAEAgAgAAAAIRAAAAEIgAAACAQAQAAAAEIgAAACAQAQAAAIEIAAAACEQAAABAIAIAAAACEQAAABCIAAAAgEAEAAAABCIAAAAgEAEAAACBCAAAAAhEAAAAQCACAAAAAhEAAAAQiAAAAIBABAAAAAQiAAAAIBABAAAAgQgAAAAIRAAAAEAgAgAAAAIRAAAAEIgAAACAQAQAAAAEIgAAACAQAQAAAIEIAAAACEQAAABAIAIAAAACEQAAABCIAAAAgEAEAAAABCIAAAAgEAEAAACBCAAAAAhEAAAAQCACAAAAAhEAAAAQiAAAAIBABAAAAAQiAAAAIBABAAAAgQgAAAAIRAAAAEAgAgAAAAIRAAAAEIgAAACAQAQAAAAEIgAAACAQAQAAAIEIAAAACEQAAABAIAIAAAACEQAAABCIAAAAgEAEAAAABCIAAAAgEAEAAACBCAAAAAhEAAAAQCACAAAAAhEAAAAQiAAAAIBABAAAAAQiAAAAIBABAAAAgQgAAAAIRAAAAEAgAgAAAAIRAAAAEIgAAACAQAQAAAAEIgAAACAQAQAAAIEIAAAACEQAAABAIAIAAAACEQAAABCIAAAAgEAEAAAABCIAAAAgEAEAAACBCAAAAAhEAAAAQCACAAAAAhEAAAAQiAAAAIBABAAAAAQiAAAAIBABAAAAgQgAAAAIRAAAAEAgAgAAAAIRAAAAEIgAAACAQAQAAAAEIgAAACAQAQAAAIEIAAAACEQAAABAIAIAAAACEQAAABCIAAAAgEAEAAAABCIAAAAgEAEAAACBCAAAAAhEAAAAQCACAAAAAhEAAAAQiAAAAIBABAAAAAQiAAAAIBABAAAAgQgAAAAIRAAAAEAgAgAAAAIRAAAAEIgAAACAQAQAAAAEIgAAACAQAQAAAIEIAAAACEQAAABAIAIAAAACEQAAABCIAAAAgEAEAAAABCIAAAAgEAEAAACBCAAAAAhEAAAAQCACAAAAAhEAAAAQiAAAAIBABAAAAAQiAAAAIBABAAAAgQgAAAAIRAAAAEAgAgAAAAIRAAAAEIgAAACAQAQAAAAEIgAAACAQAQAAAIEIAAAACEQAAABAIAIAAAACEQAAABCIAAAAgEAEAAAABCIAAAAgEAEAAACBCAAAAAhEAAAAQCACAAAAAhEAAAAQiAAAAIBABAAAAAQiAAAAIBABAAAAgQgAAAAIRAAAAEAgAgAAAAIRAAAAEIgAAACAQAQAAAAEIgAAACAQAQAAAIEIAAAACEQAAABAIAIAAAACEQAAABCIAAAAgEAEAAAABCIAAAAgEAEAAACBCAAAAAhEAAAAQCACAAAAAhEAAAAQiAAAAIBABAAAAAQiAAAAIBABAAAAgQgAAAAIRAAAAEAgAgAAAAIRAAAAEIgAAACAQAQAAAAEIgAAACAQAQAAAIEIAAAACEQAAABAIAIAAAACEQAAABCIAAAAgEAEAAAABCIAAAAgEAEAAACBCAAAAAhEAAAAQCACAAAAAhEAAAAQiAAAAIBABAAAAAQiAAAAIBABAAAAgQgAAAAIRAAAAEAgAgAAAAIRAAAAEIgAAACAQAQAAAAEIgAAACAQAQAAAIEIAAAACEQAAABAIAIAAAACEQAAABCIAAAAgEAEAAAABCIAAAAgEAEAAACBCAAAAAhEAAAAQCACAAAAAhEAAAAQiAAAAIBABAAAAAQiAAAAIBABAAAAgQgAAAAIRAAAAEAgAgAAAAIRAAAAEIgAAACAQAQAAAAEIgAAACAQAQAAAIEIAAAACEQAAABAIAIAAAACEQAAABCIAAAAgEAEAAAABCIAAAAgEAEAAACBCAAAAAhEAAAAQCACAAAAAhEAAAAQiAAAAIBABAAAAAQiAAAAIBABAAAAgQgAAAAIRAAAAEAgAgAAAAIRAAAAEIgAAACAQAQAAAAEIgAAACAQAQAAAIEIAAAACEQAAABAIAIAAAACEQAAABCIAAAAgEAEAAAABCIAAAAgEAEAAACBCAAAAAhEAAAAQCACAAAAAhEAAAAQiAAAAIBABAAAAAQiAAAAIBABAAAAgQgAAAAIRAAAAEAgAgAAAAIRAAAAEIgAAACAQAQAAAAEIgAAACAQAQAAAIEIAAAACEQAAABAIAIAAAACEQAAABCIAAAAgEAEAAAABCIAAAAgEAEAAACBCAAAAAhEAAAAQCACAAAAAhEAAAAQiAAAAIBABAAAAAQiAAAAIBABAAAAgQgAAAAIRAAAAEAgAgAAAAIRAAAAEIgAAACAQAQAAAAEIgAAACAQAQAAAIEIAAAACEQAAABAIAIAAAACEQAAABCIAAAAgED+P03g887kpY/9AAAAAElFTkSuQmCC\n\ndomain     = 'alt'\niframe_content = '<style>body { margin: 0; }</style><div>Lorem ipsum dolor sit amet, consectetur adipiscing elit.</div>'\nreference_screenshot = 'iVBORw0KGgoAAAANSUhEUgAABAAAAALkCAIAAAAeetB6AAAsmklEQVR4nO3de4DVc/748SnXXXZ1x36zy7Z085VlI7RlW+q3ulndS2kLhWyWEOkyElPoG...AIRAAAAEAgAgAAAAIRAAAAEIgAAACAQAQAAAAEIgAAACAQAQAAAIEIAAAACEQAAABAIAIAAAACEQAAABCIAAAAgED+P03g887kpY/9AAAAAElFTkSuQmCC'\nresponse   = <Responsetatus=200 body={\"value\": \"iVBORw0KGgoAAAANSUhEUgAABAAAAALkCAIAAAAeetB...QAQAAAAEIgAAACAQAQAAAIEIAAAACEQAAABAIAIAAAAC+f96b8PdM94nHQAAAABJRU5ErkJggg==\"}>\nscreenshot = 'iVBORw0KGgoAAAANSUhEUgAABAAAAALkCAIAAAAeetB6AAAr50lEQVR4nO3dfbzX8/348U/ZsLGNLoRfjK8pXUxWbIj4uvyua7ou5SITc5FRKulKoisWU...ABAAAAgQgAAAAIRAAAAEAgAgAAAAIRAAAAEIgAAACAQAQAAAAEIgAAACAQAQAAAIEIAAAACEQAAABAIAIAAAAC+f96b8PdM94nHQAAAABJRU5ErkJggg=='\nsession    = <Session 3cec4c58-be13-4c03-9a1a-f5bfb9c8bfc4>\nurl        = <function url at 0x7f6476622c80>\n\ntests/wpt/web-platform-tests/webdriver/tests/take_screenshot/iframe.py:45: AssertionError", 
    "subtest": "test_source_origin[cross_origin]", 
    "test": "/webdriver/tests/take_screenshot/iframe.py", 
    "line": 9687, 
    "action": "test_result", 
    "expected": "PASS"
}
```

--- a/components/script/dom/node.rs
+++ b/components/script/dom/node.rs
@@ -152,8 +152,6 @@ pub struct Node {
     /// Must be sent back to the layout thread to be destroyed when this
     /// node is finalized.
     style_and_layout_data: Cell<Option<OpaqueStyleAndLayoutData>>,
-
-    unique_id: UniqueId,
 }
 
 bitflags! {
@@ -1006,7 +1004,20 @@ impl Node {
     }
 
     pub fn unique_id(&self) -> String {
-        self.unique_id.borrow().to_simple().to_string()
+        let mut rare_data = self.ensure_rare_data();
+
+        if rare_data.unique_id.is_none() {
+            let id = UniqueId::new();
+            ScriptThread::save_node_id(id.borrow().to_simple().to_string());
+            rare_data.unique_id = Some(id);
+        }
+        rare_data
+            .unique_id
+            .as_ref()
+            .unwrap()
+            .borrow()
+            .to_simple()
+            .to_string()
     }
 
     pub fn summarize(&self) -> NodeInfo {
@@ -1654,7 +1665,7 @@ impl Node {
 
     #[allow(unrooted_must_root)]
     fn new_(flags: NodeFlags, doc: Option<&Document>) -> Node {
-        let node = Node {
+        Node {
             eventtarget: EventTarget::new_inherited(),
 
             parent_node: Default::default(),
@@ -1671,13 +1682,7 @@ impl Node {
             ranges: WeakRangeVec::new(),
 
             style_and_layout_data: Cell::new(None),
-
-            unique_id: UniqueId::new(),
-        };
-
-        ScriptThread::save_node_id(node.unique_id());
-
-        node
+        }
     }
 
     // https://dom.spec.whatwg.org/#concept-node-adopt
@@ -3122,7 +3127,7 @@ impl<'a> UnbindContext<'a> {
 }
 
 /// A node's unique ID, for devtools.
-struct UniqueId {
+pub struct UniqueId {
     cell: UnsafeCell<Option<Box<Uuid>>>,
 }
 

--- a/components/script/dom/raredata.rs
+++ b/components/script/dom/raredata.rs
@@ -7,6 +7,7 @@ use crate::dom::customelementregistry::{
     CustomElementDefinition, CustomElementReaction, CustomElementState,
 };
 use crate::dom::mutationobserver::RegisteredObserver;
+use crate::dom::node::UniqueId;
 use crate::dom::shadowroot::ShadowRoot;
 use std::rc::Rc;
 
@@ -22,6 +23,8 @@ pub struct NodeRareData {
     pub containing_shadow_root: Option<Dom<ShadowRoot>>,
     /// Registered observers for this node.
     pub mutation_observers: Vec<RegisteredObserver>,
+    /// Lazily-generated Unique Id for this node.
+    pub unique_id: Option<UniqueId>,
 }
 
 #[derive(Default, JSTraceable, MallocSizeOf)]

--- a/tests/unit/script/size_of.rs
+++ b/tests/unit/script/size_of.rs
@@ -30,10 +30,10 @@ macro_rules! sizeof_checker (
 
 // Update the sizes here
 sizeof_checker!(size_event_target, EventTarget, 56);
-sizeof_checker!(size_node, Node, 184);
-sizeof_checker!(size_element, Element, 392);
-sizeof_checker!(size_htmlelement, HTMLElement, 408);
-sizeof_checker!(size_div, HTMLDivElement, 408);
-sizeof_checker!(size_span, HTMLSpanElement, 408);
-sizeof_checker!(size_text, Text, 216);
-sizeof_checker!(size_characterdata, CharacterData, 216);
+sizeof_checker!(size_node, Node, 176);
+sizeof_checker!(size_element, Element, 384);
+sizeof_checker!(size_htmlelement, HTMLElement, 400);
+sizeof_checker!(size_div, HTMLDivElement, 400);
+sizeof_checker!(size_span, HTMLSpanElement, 400);
+sizeof_checker!(size_text, Text, 208);
+sizeof_checker!(size_characterdata, CharacterData, 208);

